pkg/ee/metering/report_config_handler.go

```diff
@@ -0,0 +1,369 @@
+//go:build ee
+
+/*
+                  Kubermatic Enterprise Read-Only License
+                         Version 1.0 ("KERO-1.0”)
+                     Copyright © 2022 Kubermatic GmbH
+
+   1.	You may only view, read and display for studying purposes the source
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package metering
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	apiv1 "k8c.io/kubermatic/v2/pkg/api/v1"
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	"k8c.io/kubermatic/v2/pkg/resources"
+	k8cerrors "k8c.io/kubermatic/v2/pkg/util/errors"
+	"k8c.io/kubermatic/v2/pkg/validation"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// swagger:parameters getMeteringReportConfiguration
+type getMeteringReportConfig struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
+}
+
+// swagger:parameters deleteMeteringReportConfiguration
+type deleteMeteringReportConfig struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
+}
+
+// swagger:parameters createMeteringReportConfiguration
+type createReportConfigurationReq struct {
+	// in: body
+	Body struct {
+		Name     string `json:"name"`
+		Schedule string `json:"schedule"`
+		Interval int    `json:"interval"`
+	}
+}
+
+func (m createReportConfigurationReq) Validate() error {
+	if m.Body.Name == "" {
+		return k8cerrors.NewBadRequest("name cannot be empty.")
+	}
+
+	cronExpressionParser := validation.GetCronExpressionParser()
+	if _, err := cronExpressionParser.Parse(m.Body.Schedule); err != nil {
+		return k8cerrors.NewBadRequest("invalid cron expression format: %s", m.Body.Schedule)
+	}
+
+	if m.Body.Interval < 1 {
+		return k8cerrors.NewBadRequest("interval value cannot be smaller than 1.")
+	}
+
+	return nil
+}
+
+// swagger:parameters updateMeteringReportConfiguration
+type updateReportConfigurationReq struct {
+	// in: path
+	// required: true
+	Name string `json:"report_configuration_name"`
```

Comment:
please change json for `name`

Reply:
Oh, I though you were talking about the variable naming. My bad, will update the json.